eva-accession-release-automation/run_release_in_embassy/release_common_utils.py

```diff
@@ -58,27 +60,47 @@ def get_bgzip_tabix_commands_for_file(bgzip_path, tabix_path, file):
+    return os.path.join(species_release_folder, assembly_accession, "{0}_{1}.vcf".format(assembly_accession,
+                                                                                         vcf_file_category))
 
 
-def get_unsorted_release_vcf_file_name(release_folder, assembly_accession, vcf_file_category):
-    vcf_file_path = get_release_vcf_file_name(release_folder, assembly_accession, vcf_file_category)
+def get_unsorted_release_vcf_file_name(species_release_folder, assembly_accession, vcf_file_category):
+    vcf_file_path = get_release_vcf_file_name(species_release_folder, assembly_accession, vcf_file_category)
     filename = os.path.basename(vcf_file_path)
     return vcf_file_path.replace(filename, filename.replace(".vcf", "_unsorted.vcf"))
 
 
-def get_release_text_file_name(release_folder, assembly_accession, release_text_file_category):
-    return os.path.join(release_folder, assembly_accession, "{0}_{1}.txt".format(assembly_accession,
-                                                                                 release_text_file_category))
+def get_release_text_file_name(species_release_folder, assembly_accession, release_text_file_category):
+    return os.path.join(species_release_folder, assembly_accession, "{0}_{1}.txt".format(assembly_accession,
+                                                                                         release_text_file_category))
 
 
-def get_unsorted_release_text_file_name(release_folder, assembly_accession, release_text_file_category):
-    release_text_file_path = get_release_text_file_name(release_folder, assembly_accession, release_text_file_category)
+def get_unsorted_release_text_file_name(species_release_folder, assembly_accession, release_text_file_category):
+    release_text_file_path = get_release_text_file_name(species_release_folder, assembly_accession,
+                                                        release_text_file_category)
     filename = os.path.basename(release_text_file_path)
     return release_text_file_path.replace(filename, filename.replace(".txt", ".unsorted.txt"))
 
 
-def get_release_db_name_in_tempmongo_instance(assembly_accession):
-    return "acc_" + assembly_accession.replace(".", "_")
+def get_release_db_name_in_tempmongo_instance(taxonomy_id):
+    return "acc_" + str(taxonomy_id)
+
+
+@retry(exceptions=(ConnectionError, requests.RequestException), logger=logger,
+       tries=4, delay=2, backoff=1.2, jitter=(1, 3))
+def json_request(url: str, payload: dict = None, method=requests.get) -> dict:
+    """Makes a request of a specified type (by default GET) with the specified URL and payload, attempts to parse the
+    result as a JSON string and return it as a dictionary, on failure raises an exception."""
+    result = method(url, data=payload)
+    result.raise_for_status()
+    return result.json()
+
+
+def get_ensembl_scientific_name(taxonomy_id):
```

Comment:
Why get the scientific name from ensembl when it is already present in the `release-species-inventory-table` ?  There is a slight risk that ensembl does not support that species and won't return the name.

Reply:
I have not encountered any issues with any of the taxonomies lined up for this release (see **test_ensembl_scientific_name_resolution.sh** in the eva-tasks folder for this ticket)  but if there is an issue with resolving the scientific name for a species, we might want to know about such cases and make a manual decision